From 5ca19a676cdf84a874f8b736db55f8f705e6188f Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Wed, 18 Nov 2020 16:01:34 +0100
Subject: [PATCH] Remove uart1 dts reference and change pad for UART1_RXD

UART1 is configured as rs-485, but it will be used as uart
wake-up source as described in the imx documentation.
For this we have to set the correct field values for the pad
and remove uart1 rs-485 configuration

Upstream-status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo <vicentiu@balena.io>
---
 arch/arm64/boot/dts/compulab/sb-imx8.dtsi | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/sb-imx8.dtsi b/arch/arm64/boot/dts/compulab/sb-imx8.dtsi
index 56fb9a7..7506134 100644
--- a/arch/arm64/boot/dts/compulab/sb-imx8.dtsi
+++ b/arch/arm64/boot/dts/compulab/sb-imx8.dtsi
@@ -47,7 +47,7 @@
 		};
 		pinctrl_uart1_rs485_hdx: uart1rs485hdxgrp {
 			fsl,pins = <
-				MX8MQ_IOMUXC_UART1_RXD_UART1_DCE_RX	0x79
+				MX8MQ_IOMUXC_UART1_RXD_GPIO5_IO22	0x107
 				MX8MQ_IOMUXC_UART1_TXD_UART1_DCE_TX	0x79
 			>;
 		};
@@ -163,17 +163,6 @@ sbc_i2c_lvds: &i2c2 {
 	status = "okay";
 };
 
-&uart1 { /* rs-485 */
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1_rs485_hdx>;
-	assigned-clocks = <&clk IMX8MQ_CLK_UART1>;
-	assigned-clock-parents = <&clk IMX8MQ_CLK_25M>;
-	linux,rs485-enabled-at-boot-time;
-	rs485-rts-active-high;
-	rts-gpio = <&pca9555 6 0>;
-	status = "disabled";
-};
-
 &mipi_csi_1 {
     #address-cells = <1>;
     #size-cells = <0>;
-- 
2.7.4

